vortex-file/src/pruning.rs

```diff
@@ -2,54 +2,57 @@ use std::sync::Arc;
 
 use vortex_array::ArrayRef;
 use vortex_array::arrays::{ConstantArray, StructArray};
-use vortex_array::stats::{Stat, StatsSet};
+use vortex_array::stats::{Stat, StatsProvider, StatsSet};
 use vortex_array::validity::Validity;
-use vortex_dtype::{Field, StructFields};
-use vortex_error::VortexResult;
-use vortex_expr::pruning::{PruningPredicate, access_path_stat_field_name};
+use vortex_dtype::{Field, FieldName, FieldPath, StructFields};
+use vortex_error::{VortexResult, vortex_bail};
+use vortex_expr::pruning::field_path_stat_field_name;
 use vortex_scalar::Scalar;
+use vortex_utils::aliases::hash_map::HashMap;
+use vortex_utils::aliases::hash_set::HashSet;
 
-pub fn extract_relevant_stat_as_struct_row(
-    predicate: &PruningPredicate,
-    stats_set: &Arc<[StatsSet]>,
+pub fn extract_relevant_file_stat_as_struct_row(
```

Comment:
```suggestion
pub fn extract_relevant_file_stats_as_struct_row(
```

Right? It extracts a variable number of stats.

Reply:
one row, but many stats, ye